core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -65,7 +67,7 @@ abstract class NarrowDependency[T](_rdd: RDD[T]) extends Dependency[T] {
  * @param mapSideCombine whether to perform partial aggregation (also known as map-side combine)
  */
 @DeveloperApi
-class ShuffleDependency[K, V, C](
+class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
```

Comment:
According [to the source for DeveloperApi](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/annotation/DeveloperApi.java#L23), a Developer API is unstable and can change between minor releases.